client.go

```diff
@@ -680,6 +682,32 @@ func (c *Client) Mkdir(path string) error {
 	}
 }
 
+// MkdirAll creates a directory named path, along with any necessary parents,
+// and returns nil, or else returns an error.
+// If path is already a directory, MkdirAll does nothing and returns nil.
+// If path contains a regular file, an error is returned
+func (c *Client) MkdirAll(path string) error {
+	parts := ""
+	for _, p := range strings.Split(path, "/") {
+		if p == "" {
+			continue
+		}
+		parts += "/" + p
+		dir, err := c.Stat(parts)
+		if err == nil {
+			if !dir.IsDir() {
+				return fmt.Errorf("Found a non-directory file on path: %s", parts)
```

Comment:
One should return the same error as os.MkdirAll here, `&PathError{"mkdir", parts, syscall.ENOTDIR}`